 * Don't second-guess the Debian dependencies by erroring out when a
   different version of OpenSSL is found.

-- Steve Langasek <vorlon@debian.org>  Sun, 21 Oct 2007 00:58:07 -0700
 
unchanged:
--- ntp-4.2.4p4+dfsg.orig/ntpd/ntpd.c
+++ ntp-4.2.4p4+dfsg/ntpd/ntpd.c
@@ -506,15 +506,6 @@
 	}
 #endif
 
-#ifdef OPENSSL
-	if ((SSLeay() ^ OPENSSL_VERSION_NUMBER) & ~0xff0L) {
-		msyslog(LOG_ERR,
-		    "ntpd: OpenSSL version mismatch. Built against %lx, you have %lx\n",
-		    OPENSSL_VERSION_NUMBER, SSLeay());
-		exit(1);
-	}
-#endif
-
 	/* getstartup(argc, argv); / * startup configuration, may set debug */
 
 #ifdef DEBUG
only in patch2:
unchanged:
--- ntp-4.2.4p4+dfsg.orig/util/ntp-keygen.c
+++ ntp-4.2.4p4+dfsg/util/ntp-keygen.c
@@ -255,20 +255,8 @@
 #endif
 
 #ifdef OPENSSL
-	/*
-	 * OpenSSL version numbers: MNNFFPPS: major minor fix patch status
-	 * We match major, minor, fix and status (not patch)
-	 */
-	if ((SSLeay() ^ OPENSSL_VERSION_NUMBER) & ~0xff0L) {
-		fprintf(stderr,
-		    "OpenSSL version mismatch. Built against %lx, you have %lx\n",
-		    OPENSSL_VERSION_NUMBER, SSLeay());
-		return (-1);
-
-	} else {
-		fprintf(stderr,
-		    "Using OpenSSL version %lx\n", SSLeay());
-	}
+	fprintf(stderr,
+	    "Using OpenSSL version %lx\n", SSLeay());
 #endif /* OPENSSL */
 
 	/*
